modules/data.remote/inst/satellitetools/biophys_xarray.py

```diff
@@ -0,0 +1,408 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon May 11 14:34:08 2020
+
+@author: Olli Nevalainen (olli.nevalainen@fmi.fi),
+ Finnish Meteorological Institute)
+
+Olli's python implementation of ESA SNAP s2toolbox biophysical processor and
+computation of vegetation indices.
+See ATBD at https://step.esa.int/docs/extra/ATBD_S2ToolBox_L2B_V1.1.pdf
+And java source code at
+https://github.com/senbox-org/s2tbx/tree/master/s2tbx-biophysical/src/main/java/org/esa/s2tbx/biophysical
+
+Caveats
+Currently changes out of bounds inputs and outputs to nan (or min or max value
+if output wihtin tolerance). Maybe output flagging information as well ( i.e.
+diffferent flags input and output out of bounds).
+
+Convex hull input checking currently disabled. It's computationally slow and
+ not sure of its benefits. Better to filter out bad data based on L2A quality
+ info/classification\
+    and hope averaging removes some bad pixels.
+"""
+
+import requests
+import io
+import numpy as np
+import xarray as xr
+
+# url to Sentinel 2 Toolbox's auxdata
+# This base_url points towards the original toolbox(not the one created by Olli)
+base_url = "https://raw.githubusercontent.com/senbox-org/s2tbx/master/s2tbx-biophysical/src/main/resources/auxdata/2_1/{}/{}"
+
+
+def get_fromurl(var, pattern):
+    """
+    Fetches the contents of a text file from the base url and stores it in a ndarray.
+
+    Author: Ayush Prasad
+
+    Parameters
+    ----------
+    var (str) -- type of the product, one of FAPAR, FCOVER, LAI, LAI_Cab and LAI_Cw.
+    pattern (str) -- name of the file excluding the initial variable part.
+
+    Returns
+    -------
+    ndarray -- loaded with the contents of the text file.
+    """
+    # attach variable and file name to the base url
+    res_url = base_url.format(var, str(var) + "%s" % str(pattern))
+    # make a GET request to the url to fetch the data.
+    res_url = requests.get(res_url)
+    # check the HTTP status code to see if any error has occured.
+    res_url.raise_for_status()
+    # store the contents of the url in an in-memory buffer and use it to load the ndarray.
+    return np.loadtxt(io.BytesIO(res_url.content), delimiter=",")
+
+
+# Read SNAP Biophysical processor neural network parameters
+nn_params = {}
+for var in ["FAPAR", "FCOVER", "LAI", "LAI_Cab", "LAI_Cw"]:
```

Comment:
I can make this part specifically for LAI instead of the generalized approach.

Reply:
As said in the previous comment, let's keep it this way for now